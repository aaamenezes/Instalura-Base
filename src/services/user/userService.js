import { isStagingEnv } from '../../infra/env/isStagingEnv'
import { HttpClient } from '../../infra/http/HttpClient'
import { authService } from '../auth/authService'

const BASE_URL = isStagingEnv
  // URL de DEV
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
  // URL DE PROD
  : 'https://instalura-api-git-master-omariosouto.vercel.app'

export const userService = {
  async getProfilePage() {
    const url = `${ BASE_URL }/api/users/posts`
    try {
      const token = await authService(context).getToken()
      const response = await HttpClient(url, {
        headers: {
          Authorization: `Bearer ${ token }`
        }
      })
      return {
        user: {
          totalLikes: 100
        },
        posts: response.data
      }
    } catch (error) {
      throw new Error('NÃ£o consegui trazer os posts')
    }
  }
}
