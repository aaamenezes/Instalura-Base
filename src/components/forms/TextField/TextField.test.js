import React from 'react'
import user from '@testing-library/user-event'
import { render, screen } from '../../../infra/test/testUtils'
import TextField from '.'

describe('<TextField />', () => {
  test('renders component', () => {
    render(
      <TextField
        placeholder='Nome'
        value='Mario'
        onChange={() => {}}
        name='name'
      />
    )
    // screen.debug()
    const textField = screen.getByPlaceholderText(/nome/i)
    expect(textField).toMatchSnapshot()
  })

  describe('when field is valid', () => {
    describe('and user is typing', () => {
      test('the value must be updated', () => {
        const onChangeMock = jest.fn()
        render(
          <TextField
            placeholder='Nome'
            value='Andre'
            onChange={onChangeMock}
            name='nome'
            isTouched
          />
        )
        const inputNome = screen.getByPlaceholderText(/nome/i)
        user.type(inputNome, 'devsoutinho')
        expect(onChangeMock).toHaveBeenCalledTimes(11)
      })
    })
  })

  describe('when field is invalid', () => {
    // Elemento tenha o span de texto
    test('display the respective error mesage', () => {
      render(
        <TextField
          placeholder='Email'
          value='@gmail.com'
          onChange={() => {}}
          name='email'
          isTouched
          error='campo obrigatório'
        />
      )
      expect(screen.getByPlaceholderText(/email/i)).toHaveValue('@gmail.com')
      expect(screen.getByRole('alert')).toHaveTextContent('campo obrigatório')

      const inputEmail = screen.getByPlaceholderText(/email/i)
      expect(inputEmail).toMatchSnapshot()
    })
  })
})
