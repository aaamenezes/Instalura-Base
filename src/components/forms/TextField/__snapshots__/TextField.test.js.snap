// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #CCC;
  padding: 12px 16px;
  outline: 0;
  border-radius: 8px;
}

<input
  class="c0 c1"
  name="name"
  placeholder="Nome"
  type="text"
  value="Mario"
/>
`;

exports[`<TextField /> when field is invalid display the respective error mesage 1`] = `
.c0 {
  font-size: 16px;
  font-weight: 400;
  line-height: 1.25;
}

.c1 {
  width: 100%;
  border: 1px solid #CCC;
  padding: 12px 16px;
  outline: 0;
  border-radius: 8px;
  border-color: #DC3545;
}

.c1 + span {
  color: #DC3545 font-size:11px;
}

<input
  class="c0 c1"
  name="email"
  placeholder="Email"
  type="text"
  value="@gmail.com"
/>
`;
